cpp/CMakeLists.txt

```diff
@@ -174,127 +174,312 @@ set(CUGRAPH_SOURCES
     src/detail/collect_local_vertex_values.cu
     src/detail/groupby_and_count.cu
     src/detail/collect_comm_wrapper.cu
-    src/sampling/random_walks_mg.cu
-    src/community/detail/common_methods_mg.cu
-    src/community/detail/common_methods_sg.cu
-    src/community/detail/refine_sg.cu
-    src/community/detail/refine_mg.cu
-    src/community/edge_triangle_count_sg.cu
-    src/community/edge_triangle_count_mg.cu
-    src/community/detail/maximal_independent_moves_sg.cu
-    src/community/detail/maximal_independent_moves_mg.cu
+    src/sampling/random_walks_mg64.cu
+    src/sampling/random_walks_mg32.cu
+    src/sampling/random_walks_mg3264.cu
+    src/community/detail/common_methods_mg64.cu
+    src/community/detail/common_methods_mg32.cu
+    src/community/detail/common_methods_mg3264.cu
+    src/community/detail/common_methods_sg64.cu
+    src/community/detail/common_methods_sg32.cu
+    src/community/detail/common_methods_sg3264.cu
+    src/community/detail/refine_sg64.cu
+    src/community/detail/refine_sg32.cu
+    src/community/detail/refine_sg3264.cu
+    src/community/detail/refine_mg64.cu
+    src/community/detail/refine_mg32.cu
+    src/community/detail/refine_mg3264.cu
+    src/community/edge_triangle_count_sg64.cu
+    src/community/edge_triangle_count_sg32.cu
+    src/community/edge_triangle_count_sg3264.cu
+    src/community/edge_triangle_count_mg64.cu
+    src/community/edge_triangle_count_mg32.cu
+    src/community/edge_triangle_count_mg3264.cu
+    src/community/detail/maximal_independent_moves_sg64.cu
+    src/community/detail/maximal_independent_moves_sg32.cu
+    src/community/detail/maximal_independent_moves_sg3264.cu
+    src/community/detail/maximal_independent_moves_mg64.cu
+    src/community/detail/maximal_independent_moves_mg32.cu
+    src/community/detail/maximal_independent_moves_mg3264.cu
     src/detail/utility_wrappers.cu
-    src/structure/graph_view_mg.cu
+    src/structure/graph_view_mg64.cu
+    src/structure/graph_view_mg32.cu
+    src/structure/graph_view_mg3264.cu
     src/structure/remove_self_loops.cu
     src/structure/remove_multi_edges.cu
```

Comment:
This function is SG only but explicitly instantiates based on vertex_t and edge_t, no need to split?